Feature/69 Refactor 19&Me shiny app to hit risk-score calculator API

pendlrobinson opened the pull request:
Jira Ticket: https://mathematicampr.atlassian.net/browse/CWRT-69

Read for the branch to be merged. 

There are separate tickets for:
- Update "methods" tab to say that case fatality rate uses 90-day sliding window of cases and deaths [C19ANDME-53](https://mathematicampr.atlassian.net/browse/C19ANDME-53)
- Update README [C19ANDME-52](https://mathematicampr.atlassian.net/browse/C19ANDME-52)

xindyhu reviewed (changes requested):
Looking good! Thank you.
A few minor comments. Feel free to lmk if you want to discuss any.

xindyhu commented on app/global.R:
```diff
@@ -0,0 +1,66 @@
+  menni_etall_2020 = "https://www.nature.com/articles/s41591-020-0916-2",
+  simonnet_etall_2020 = "https://onlinelibrary.wiley.com/doi/full/10.1002/oby.22831?af=R", 
+  # misc
+  conjugate_priors = "https://en.wikipedia.org/wiki/Conjugate_prior",
```
Thanks @e-lipman Just saw the other PR!

xindyhu commented on app/results.R:
```diff
@@ -0,0 +1,197 @@
+               We estimate that your county has an under-reporting factor of ', underreport_factor_string, 
+               '. Taking into account the under-reporting factor, incubation period, and time from symptom onset to recovery, we estimate there are ',
+               formatDynamicString(format(round(risk$est_unreported_sick), big.mark =",")),
+               ' sick people distributed through the county who are not officially reported.'
```
Erin helped with solving this in the other PR.

xindyhu commented on app/server.R:
```diff
@@ -0,0 +1,138 @@
+  showModal(disclaimer_message)
+  
+  # function to make request to API
+  hit_api<- eventReactive(input$go, {
```
After the first hit of the API, how do you feel about making it Reactive not eventReactive so that it can respond to user actions like tick/untick, sliding up/down the number of close contacts?

pendlrobinson replied:
I think that could work. I will have to do some googling about how to implement the functionality of _After the first hit of the API making it Reactive_

pendlrobinson commented on app/results.R:
```diff
@@ -0,0 +1,197 @@
+    "ppe"= bool2char(input$ppe),
+    "conditions" = as.list(input$conditions))
+  
+  resp <- POST(request_url, add_headers("x-api-key" = Sys.getenv("X_API_KEY")), body = request_body, encode = "json")
```
I think we spoke about this at the last 19andMe meeting. I think we decided that we were not going to give out API keys. Has this changed?

pendlrobinson commented on app/results.R:
```diff
@@ -0,0 +1,197 @@
+    formatPercent(risk$hosp_risk),
+    ", the risk of requiring an ICU is ",
+    formatPercent(risk$icu_risk),
+    ", and the risk of dying is ",
```
I think "risk of not surviving" sounds more palatable. I will change the wording

margaretluo commented on README.md:
```diff
@@ -1,68 +1,29 @@
+* **Emma Pendl-Robinson, Erin Lipman** - *back-end developer*
+
+## Reviewers and Support
+* **Margaret Luo** - *Docker guru*
```
I personally don't really like the use of the word guru in this context because it feels like appropriation from Buddhist and Hindu religions. Could we use the word "expert" or "implementer" instead?

pendlrobinson replied:
Thanks Margaret for your comment, I agree. 
I think this change should be made in the readme pr (https://github.com/mathematica-mpr/covid_risk_score/pull/51)

pendlrobinson commented:
@xindyhu Thank you for your feedback. It would be great to get your approval to complete this PR and merge this hit_api to develop.

I implemented the changes you suggested:
- add how information on how people can get an API key
- add a delay to the zip code input making calls to the api (I added a validation that the zip code must be at least 5 characters long before the RShiny app will make an api request).
- remove the word "guru" 
- change "risk of dying" to "risk of not surviving"

I think some of the other changes can be done via new feature branches. I made some Jira tickets to remind me to make these additions:
- adding people's titles to the README - https://mathematicampr.atlassian.net/browse/C19ANDME-52
- fix the prompt “risk for people like you is” to make it disappear - https://mathematicampr.atlassian.net/browse/C19ANDME-58
- update the shiny code for hands and ppe to use the API outputs - https://mathematicampr.atlassian.net/browse/C19ANDME-59
